Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/QueryControllerFunctions.java

```diff
@@ -77,143 +72,146 @@ public static ResponseEntity<String> getEntity(EntryQueryService queryService, H
-			return ResponseEntity.status(HttpStatus.NOT_FOUND)
-					.body(new RestResponse(ErrorType.NotFound, "Resource not found.").toJson());
-		}
-		return result;
+		MultiMap params = request.params();
+		params.add(NGSIConstants.QUERY_PARAMETER_ATTRS, attribute);
+		return getQueryData(queryService, request, "", params, false, false, false, defaultLimit, maxLimit, check)
+				.onItem().transform(t -> {
+					if (t.getEntity().equals(emptyResult1) || t.getEntity().equals(emptyResult2)) {
+						return HttpUtils.NOT_FOUND_REPLY;
+					} else {
+						return t;
+					}
+				}).onFailure().recoverWithItem(HttpUtils::handleControllerExceptions);
 	}
 
-	private static ResponseEntity<String> getQueryData(EntryQueryService queryService, HttpServletRequest request,
-			String originalQueryParams, MultiValueMap<String, String> paramMap, boolean typeRequired,
-			boolean forceArray, boolean temporal, int defaultLimit, int maxLimit, String check) {
-		String tenantid = request.getHeader(NGSIConstants.TENANT_HEADER_FOR_INTERNAL_CHECK);
-		try {
-			logger.trace("getAllEntity() ::");
-			List<Object> linkHeaders = HttpUtils.getAtContext(request);
-			Context context = JsonLdProcessor.getCoreContextClone();
-			context = context.parse(linkHeaders, true);
-			if (originalQueryParams != null) {
-				originalQueryParams = URLDecoder.decode(originalQueryParams, NGSIConstants.ENCODE_FORMAT);
-				QueryParams qp = ParamsResolver.getQueryParamsFromUriQuery(paramMap, context, temporal, typeRequired,
-						defaultLimit, maxLimit);
-				if (qp == null) // invalid query
-					throw new ResponseException(ErrorType.InvalidRequest, "Empty query parameters");
-				qp.setTenant(tenantid);
-				qp.setCheck(check);
-				ArrayListMultimap<String, String> headers = HttpUtils.getHeaders(request);
-				QueryResult qResult;
-				try {
-					qResult = queryService.getData(qp, originalQueryParams, linkHeaders, headers, false);
-				} catch (Exception e) {
-					// logger.error(e.getMessage());
-					logger.error("Tenant for the request not found", e);
-					return ResponseEntity.status(HttpStatus.NOT_FOUND)
-							.body(new RestResponse(ErrorType.TenantNotFound, "Tenant not found.").toJson());
-				}
-				return HttpUtils.generateReply(request, qResult, forceArray, qp.getCountResult(), context, linkHeaders,
-						AppConstants.QUERY_ENDPOINT);
-
-			} else {
-				return HttpUtils
-						.handleControllerExceptions(new ResponseException(ErrorType.BadRequestData, "invalid query"));
-			}
-		} catch (Exception exception) {
-			return HttpUtils.handleControllerExceptions(exception);
+	private static Uni<RestResponse<Object>> getQueryData(EntryQueryService queryService, HttpServerRequest request,
+			String originalQueryParams, MultiMap paramMap, boolean typeRequired, boolean forceArray, boolean temporal,
+			int defaultLimit, int maxLimit, String check) {
+		logger.trace("getAllEntity() ::");
+		if (originalQueryParams == null || originalQueryParams.isEmpty()) {
```

Comment:
[Major] I think this condition is not working in case of query Entity types/Attributes API as : /ngsi-ld/v1/types/Building  So please check

Reply:
verified